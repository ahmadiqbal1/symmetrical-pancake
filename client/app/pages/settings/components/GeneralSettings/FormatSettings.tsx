import React from "react";
import { SettingsEditorPropTypes, SettingsEditorDefaultProps } from "../prop-types";
import Form from "antd/lib/form";
import Select from "antd/lib/select";
import Skeleton from "antd/lib/skeleton";
import DynamicComponent from "@/components/DynamicComponent";
import { clientConfig } from "@/services/auth";
export default function FormatSettings(props: any) {
    const { values, onChange, loading } = props;
    return (<DynamicComponent name="OrganizationSettings.FormatSettings" {...props}>
      <Form.Item label="Date Format">
        {loading ? (<Skeleton.Input style={{ width: 300 }} active/>) : (<Select value={values.date_format} onChange={value => onChange({ date_format: value })} data-test="DateFormatSelect">
            {/* @ts-expect-error ts-migrate(2741) FIXME: Property 'value' is missing in type '{ children: a... Remove this comment to see the full error message */}
            {(clientConfig as any).dateFormatList.map((dateFormat: any) => <Select.Option key={dateFormat} data-test={`DateFormatSelect:${dateFormat}`}>
              {dateFormat}
            </Select.Option>)}
          </Select>)}
      </Form.Item>
      <Form.Item label="Time Format">
        {loading ? (<Skeleton.Input style={{ width: 300 }} active/>) : (<Select value={values.time_format} onChange={value => onChange({ time_format: value })} data-test="TimeFormatSelect">
            {/* @ts-expect-error ts-migrate(2741) FIXME: Property 'value' is missing in type '{ children: a... Remove this comment to see the full error message */}
            {(clientConfig as any).timeFormatList.map((timeFormat: any) => <Select.Option key={timeFormat}>{timeFormat}</Select.Option>)}
          </Select>)}
      </Form.Item>
    </DynamicComponent>);
}
FormatSettings.propTypes = SettingsEditorPropTypes;
FormatSettings.defaultProps = SettingsEditorDefaultProps;
