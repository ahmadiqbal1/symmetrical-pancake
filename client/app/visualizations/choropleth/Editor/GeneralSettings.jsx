import { isString, map, filter } from "lodash";
import React, { useMemo } from "react";
import { useDebouncedCallback } from "use-debounce";
import * as Grid from "antd/lib/grid";
import { EditorPropTypes } from "@/visualizations/prop-types";
import { Section, Select, Input } from "@/components/visualizations/editor";

import useLoadGeoJson from "../hooks/useLoadGeoJson";
import availableMaps, { getMapUrl } from "../maps";
import { getGeoJsonFields } from "./utils";

export default function GeneralSettings({ options, data, onOptionsChange }) {
  const [geoJson, isLoadingGeoJson] = useLoadGeoJson(getMapUrl(options.mapType, options.customMapUrl));
  const geoJsonFields = useMemo(() => getGeoJsonFields(geoJson), [geoJson]);

  // While geoJson is loading - show last selected field in select
  const targetFields = isLoadingGeoJson ? filter([options.targetField], isString) : geoJsonFields;

  const [handleCustomMapUrlChange] = useDebouncedCallback(customMapUrl => {
    onOptionsChange({ customMapUrl });
  }, 200);

  return (
    <React.Fragment>
      <Section>
        <Select
          label="Map type"
          className="w-100"
          data-test="Choropleth.Editor.MapType"
          defaultValue={options.mapType}
          onChange={mapType => onOptionsChange({ mapType })}>
          {map(availableMaps, ({ name }, mapKey) => (
            <Select.Option key={mapKey} data-test={`Choropleth.Editor.MapType.${mapKey}`}>
              {name}
            </Select.Option>
          ))}
          <Select.Option key="custom" data-test="Choropleth.Editor.MapType.custom">
            Custom...
          </Select.Option>
        </Select>
      </Section>

      {options.mapType === "custom" && (
        <Section>
          <Input
            data-test="Choropleth.Editor.CustomMapUrl"
            placeholder="Custom map URL..."
            defaultValue={options.customMapUrl}
            onChange={event => handleCustomMapUrlChange(event.target.value)}
          />
        </Section>
      )}

      <Section>
        <Grid.Row gutter={15}>
          <Grid.Col span={12}>
            <Select
              label="Key column"
              className="w-100"
              data-test="Choropleth.Editor.KeyColumn"
              disabled={data.columns.length === 0}
              defaultValue={options.keyColumn}
              onChange={keyColumn => onOptionsChange({ keyColumn })}>
              {map(data.columns, ({ name }) => (
                <Select.Option key={name} data-test={`Choropleth.Editor.KeyColumn.${name}`}>
                  {name}
                </Select.Option>
              ))}
            </Select>
          </Grid.Col>
          <Grid.Col span={12}>
            <Select
              label="Target field"
              className="w-100"
              data-test="Choropleth.Editor.TargetField"
              disabled={isLoadingGeoJson || targetFields.length === 0}
              loading={isLoadingGeoJson}
              value={options.targetField}
              onChange={targetField => onOptionsChange({ targetField })}>
              {map(targetFields, field => (
                <Select.Option key={field} data-test={`Choropleth.Editor.TargetField.${field}`}>
                  {field}
                </Select.Option>
              ))}
            </Select>
          </Grid.Col>
        </Grid.Row>
      </Section>

      <Section>
        <Select
          label="Value column"
          className="w-100"
          data-test="Choropleth.Editor.ValueColumn"
          disabled={data.columns.length === 0}
          defaultValue={options.valueColumn}
          onChange={valueColumn => onOptionsChange({ valueColumn })}>
          {map(data.columns, ({ name }) => (
            <Select.Option key={name} data-test={`Choropleth.Editor.ValueColumn.${name}`}>
              {name}
            </Select.Option>
          ))}
        </Select>
      </Section>
    </React.Fragment>
  );
}

GeneralSettings.propTypes = EditorPropTypes;
