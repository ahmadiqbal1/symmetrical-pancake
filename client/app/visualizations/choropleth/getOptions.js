import { isNil, merge, first, keys, get } from "lodash";
import ColorPalette from "./ColorPalette";
import availableMaps from "./maps";

const defaultMap = first(keys(availableMaps));

const DEFAULT_OPTIONS = {
  mapType: defaultMap,
  customMapUrl: null,
  keyColumn: null,
  targetField: null,
  valueColumn: null,
  clusteringMode: "e",
  steps: 5,
  valueFormat: "0,0.00",
  noValuePlaceholder: "N/A",
  colors: {
    min: ColorPalette["Light Blue"],
    max: ColorPalette["Dark Blue"],
    background: ColorPalette.White,
    borders: ColorPalette.White,
    noValue: ColorPalette["Light Gray"],
  },
  legend: {
    visible: true,
    position: "bottom-left",
    alignText: "right",
  },
  tooltip: {
    enabled: true,
    template: "<b>{{ @@name }}</b>: {{ @@value }}",
  },
  popup: {
    enabled: true,
    template: "Country: <b>{{ @@name_long }} ({{ @@iso_a2 }})</b>\n<br>\nValue: <b>{{ @@value }}</b>",
  },
};

export default function getOptions(options) {
  const result = merge({}, DEFAULT_OPTIONS, options);
  // Both renderer and editor always provide new `bounds` array, so no need to clone it here.
  // Keeping original object also reduces amount of updates in components
  result.bounds = get(options, "bounds");

  if (isNil(availableMaps[result.mapType]) && result.mapType !== "custom") {
    result.mapType = defaultMap;
  }

  // backward compatibility
  if (!isNil(result.countryCodeColumn)) {
    result.keyColumn = result.countryCodeColumn;
    delete result.countryCodeColumn;
  }
  if (!isNil(result.countryCodeType)) {
    result.targetField = result.countryCodeType;
    delete result.countryCodeType;
  }

  return result;
}
