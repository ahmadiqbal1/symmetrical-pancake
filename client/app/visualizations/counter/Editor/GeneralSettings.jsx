import React from "react";
import { Section, Input, ContextHelp } from "@/components/visualizations/editor";
import { EditorPropTypes } from "@/visualizations/prop-types";

export default function GeneralSettings({ options, visualizationName, onOptionsChange }) {
  return (
    <React.Fragment>
      <Section>
        <Input
          layout="horizontal"
          label="Counter Label"
          className="w-100"
          data-test="Counter.CounterLabel"
          defaultValue={options.counterLabel}
          placeholder={visualizationName}
          onChange={e => onOptionsChange({ counterLabel: e.target.value })}
        />
      </Section>

      <Section>
        <Input
          layout="horizontal"
          label={
            <>
              Number Format <ContextHelp.NumberFormatSpecs />
            </>
          }
          className="w-100"
          data-test="Counter.NumberFormat"
          defaultValue={options.numberFormat}
          onChange={e => onOptionsChange({ numberFormat: e.target.value })}
        />
      </Section>

      <Section>
        <Input
          layout="horizontal"
          label="Formatting Decimal Character"
          className="w-100"
          data-test="Counter.DecimalCharacter"
          defaultValue={options.stringDecChar}
          onChange={e => onOptionsChange({ stringDecChar: e.target.value })}
        />
      </Section>

      <Section>
        <Input
          layout="horizontal"
          label="Formatting Thousands Separator"
          className="w-100"
          data-test="Counter.ThousandsSeparator"
          defaultValue={options.stringThouSep}
          onChange={e => onOptionsChange({ stringThouSep: e.target.value })}
        />
      </Section>
    </React.Fragment>
  );
}

GeneralSettings.propTypes = EditorPropTypes;
