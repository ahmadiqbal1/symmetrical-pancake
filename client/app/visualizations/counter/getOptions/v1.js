import { extend } from "lodash";

const schemaVersion = 1;

const defaultOptions = {
  counterLabel: "",
  counterColName: "counter",
  rowNumber: 1,
  targetColName: null,
  targetRowNumber: 1,
  stringDecimal: 0,
  stringDecChar: ".",
  stringThouSep: ",",
  stringPrefix: null,
  stringSuffix: null,
  formatTargetValue: false,
  countRow: false,
};

export default function getOptions(options) {
  return extend({}, defaultOptions, options, { schemaVersion });
}
