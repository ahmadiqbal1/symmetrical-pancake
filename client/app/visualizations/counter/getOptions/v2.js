import { isEmpty, isFinite, merge, get, pick, padEnd, toString, includes, map } from "lodash";
import getOptionsV1 from "./v1";

const schemaVersion = 2;

const defaultOptions = {
  schemaVersion,

  counterLabel: "",

  stringDecChar: ".",
  stringThouSep: ",",
  numberFormat: "0,0",

  primaryValue: {
    show: true,
    type: "rowValue",
    column: "counter",
    rowNumber: 1,
    displayFormat: "{{ @@value_formatted }}",
    showTooltip: true,
    tooltipFormat: "{{ @@value }}",
  },
  secondaryValue: {
    show: true,
    type: "unused",
    column: null,
    rowNumber: 1,
    displayFormat: "({{ @@value_formatted }})",
    showTooltip: true,
    tooltipFormat: "{{ @@value }}",
  },
};

function migrateFromV1(options) {
  options = getOptionsV1(options);
  const result = pick(options, ["counterLabel", "stringDecChar", "stringThouSep"]);

  result.numberFormat = "0,0.000";
  if (isFinite(options.stringDecimal) && options.stringDecimal >= 0) {
    result.numberFormat = "0,0";
    if (options.stringDecimal > 0) {
      const decimals = padEnd("", options.stringDecimal, "0");
      result.numberFormat = `${result.numberFormat}.${decimals}`;
    }
  }

  const prefix = toString(options.stringPrefix);
  const suffix = toString(options.stringSuffix);

  result.primaryValue = {
    show: true,
    type: options.countRow ? "countRows" : "rowValue",
    column: options.counterColName,
    rowNumber: options.rowNumber,
    displayFormat: `${prefix}{{ @@value_formatted }}${suffix}`,
  };

  result.secondaryValue = {
    show: true,
    type: options.targetColName ? "rowValue" : "unused",
    column: options.targetColName,
    rowNumber: options.targetRowNumber,
    displayFormat: options.formatTargetValue ? `(${prefix}{{ @@value_formatted }}${suffix})` : "({{ @@value }})",
  };

  return result;
}

export default function getOptions(options, { columns }) {
  const currentSchemaVersion = get(options, "schemaVersion", isEmpty(options) ? schemaVersion : 0);
  if (currentSchemaVersion < schemaVersion) {
    options = migrateFromV1(options);
  }

  const optionsWithDefaults = merge({}, defaultOptions, options, { schemaVersion });
  const columnNameUpdates = {};

  const columnNames = map(columns, col => col.name);
  if (!includes(columnNames, get(optionsWithDefaults, "primaryValue.column"))) {
    columnNameUpdates.primaryValue = { column: null };
  }
  if (!includes(columnNames, get(optionsWithDefaults, "secondaryValue.column"))) {
    columnNameUpdates.secondaryValue = { column: null };
  }

  return merge({}, optionsWithDefaults, columnNameUpdates);
}
