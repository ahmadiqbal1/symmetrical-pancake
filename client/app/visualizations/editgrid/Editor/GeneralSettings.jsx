import { map } from "lodash";
import React from "react";
import * as Grid from "antd/lib/grid";
import Select from "antd/lib/select";
import Switch from "antd/lib/switch";
import { EditorPropTypes } from "@/visualizations";

export default function GeneralSettings({
  options,
  onOptionsChange,
  schema,
  tableColumns,
  commonColumns,
}) {
  function updateColumns(columnName, key, value, optionKey) {
    const changedColumn = { ...options.columns[columnName], [key]: value };
    onOptionsChange({
      columns: { ...options.columns, [columnName]: changedColumn },
      [optionKey]: columnName,
    });
  }

  return (
    <React.Fragment>
      <Grid.Row type="flex" align="middle" className="m-b-10">
        <Grid.Col span={12}>
          <label htmlFor="edit-table">Table to be Edited</label>
        </Grid.Col>
        <Grid.Col span={12}>
          <Select
            id="edit-table"
            className="w-100"
            data-test="EditGrid.General.EditTable"
            defaultValue={options.editTable}
            onChange={(editTable) =>
              onOptionsChange({ editTable, columns: {} })
            }
            showSearch
            filterOption={(input, option) =>
              option.props.children
                .toLowerCase()
                .indexOf(input.toLowerCase()) >= 0
            }
          >
            {map(schema, (table) => (
              <Select.Option
                key={table.name}
                data-test={"EditGrid.General.EditTable." + table.name}
              >
                {table.name}
              </Select.Option>
            ))}
          </Select>
        </Grid.Col>
      </Grid.Row>

      <Grid.Row type="flex" align="middle" className="m-b-10">
        <Grid.Col span={12}>
          <label htmlFor="edit-primary-key">Primary Key</label>
        </Grid.Col>
        <Grid.Col span={12}>
          <Select
            id="edit-primary-key"
            className="w-100"
            data-test="EditGrid.General.PrimaryKey"
            defaultValue={options.primaryKey}
            onChange={(primaryKey) =>
              updateColumns(primaryKey, "edit", false, "primaryKey")
            }
            showSearch
            filterOption={(input, option) =>
              option.props.children
                .toLowerCase()
                .indexOf(input.toLowerCase()) >= 0
            }
          >
            {map(commonColumns, (col) => (
              <Select.Option
                key={col.name}
                data-test={"EditGrid.General.PrimaryKey." + col.name}
              >
                {col.name}
              </Select.Option>
            ))}
          </Select>
        </Grid.Col>
      </Grid.Row>

      <Grid.Row type="flex" align="middle" className="m-b-10">
        <Grid.Col span={8}>
          <label
            className="d-flex align-items-center"
            htmlFor="edit-is-marking-user"
          >
            <Switch
              id="edit-is-marking-user"
              data-test="EditGrid.General.IsMarkingUser"
              defaultChecked={options.isMarkingUser}
              onChange={(isMarkingUser) => onOptionsChange({ isMarkingUser })}
            />
            <span className="m-l-10">Mark User</span>
          </label>
        </Grid.Col>
        <Grid.Col span={8}>
          <label
            className="d-flex align-items-center"
            htmlFor="edit-is-versioning"
          >
            <Switch
              id="edit-is-versioning"
              data-test="EditGrid.General.IsVersioning"
              defaultChecked={options.isVersioning}
              onChange={(isVersioning) => onOptionsChange({ isVersioning })}
            />
            <span className="m-l-10">Version Data</span>
          </label>
        </Grid.Col>
        <Grid.Col span={8}>
          <label
            className="d-flex align-items-center"
            htmlFor="edit-is-snapshotting"
          >
            <Switch
              id="edit-is-snapshotting"
              data-test="EditGrid.General.IsSnapshotting"
              defaultChecked={options.isSnapshotting}
              onChange={(isSnapshotting) => onOptionsChange({ isSnapshotting })}
            />
            <span className="m-l-10">Snapshot Data</span>
          </label>
        </Grid.Col>
      </Grid.Row>

      {options.isMarkingUser ? (
        <Grid.Row type="flex" align="middle" className="m-b-10">
          <Grid.Col span={12}>
            <label htmlFor="edit-modified-by">Modified By User Column</label>
          </Grid.Col>
          <Grid.Col span={12}>
            <Select
              id="edit-modified-by"
              className="w-100"
              data-test="EditGrid.General.ModifiedBy"
              defaultValue={options.modifiedBy}
              onChange={(modifiedBy) =>
                updateColumns(modifiedBy, "edit", false, "modifiedBy")
              }
              showSearch
              filterOption={(input, option) =>
                option.props.children
                  .toLowerCase()
                  .indexOf(input.toLowerCase()) >= 0
              }
            >
              {map(tableColumns, (col) => (
                <Select.Option
                  key={col}
                  data-test={"EditGrid.General.ModifiedBy." + col}
                >
                  {col}
                </Select.Option>
              ))}
            </Select>
          </Grid.Col>
        </Grid.Row>
      ) : null}

      {options.isVersioning ? (
        <Grid.Row type="flex" align="middle" className="m-b-10">
          <Grid.Col span={12}>
            <label htmlFor="edit-updated-at">Update Time Column</label>
          </Grid.Col>
          <Grid.Col span={12}>
            <Select
              id="edit-updated-at"
              className="w-100"
              data-test="EditGrid.General.UpdatedAt"
              defaultValue={options.updatedAt}
              onChange={(updatedAt) =>
                updateColumns(updatedAt, "edit", false, "updatedAt")
              }
              showSearch
              filterOption={(input, option) =>
                option.props.children
                  .toLowerCase()
                  .indexOf(input.toLowerCase()) >= 0
              }
            >
              {map(tableColumns, (col) => (
                <Select.Option
                  key={col}
                  data-test={"EditGrid.General.UpdatedAt." + col}
                >
                  {col}
                </Select.Option>
              ))}
            </Select>
          </Grid.Col>
        </Grid.Row>
      ) : null}

      {options.isSnapshotting ? (
        <Grid.Row type="flex" align="middle" className="m-b-10">
          <Grid.Col span={11} style={{ marginRight: 10 }}>
            <label htmlFor="edit-start-time">Start Time Column</label>
            <Select
              id="edit-start-time"
              className="w-100"
              data-test="EditGrid.General.StartTime"
              defaultValue={options.startTime}
              onChange={(startTime) =>
                updateColumns(startTime, "edit", false, "startTime")
              }
              showSearch
              filterOption={(input, option) =>
                option.props.children
                  .toLowerCase()
                  .indexOf(input.toLowerCase()) >= 0
              }
            >
              {map(tableColumns, (col) => (
                <Select.Option
                  key={col}
                  data-test={"EditGrid.General.StartTime." + col}
                >
                  {col}
                </Select.Option>
              ))}
            </Select>
          </Grid.Col>
          <Grid.Col span={11}>
            <label htmlFor="edit-end-time">End Time Column</label>
            <Select
              id="edit-end-time"
              className="w-100"
              data-test="EditGrid.General.EndTime"
              defaultValue={options.endTime}
              onChange={(endTime) =>
                updateColumns(endTime, "edit", false, "endTime")
              }
              showSearch
              filterOption={(input, option) =>
                option.props.children
                  .toLowerCase()
                  .indexOf(input.toLowerCase()) >= 0
              }
            >
              {map(tableColumns, (col) => (
                <Select.Option
                  key={col}
                  data-test={"EditGrid.General.EndTime." + col}
                >
                  {col}
                </Select.Option>
              ))}
            </Select>
          </Grid.Col>
        </Grid.Row>
      ) : null}

      <label className="d-flex align-items-center" htmlFor="edit-allow-delete">
        <Switch
          id="edit-allow-delete"
          data-test="EditGrid.General.AllowDelete"
          defaultChecked={options.allowDelete}
          onChange={(allowDelete) => onOptionsChange({ allowDelete })}
        />
        <span className="m-l-10">Allow Deletion of Rows</span>
      </label>
    </React.Fragment>
  );
}

GeneralSettings.propTypes = EditorPropTypes;
