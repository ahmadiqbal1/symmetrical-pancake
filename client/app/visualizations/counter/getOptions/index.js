import getOptions from "./v2";

export default getOptions;
