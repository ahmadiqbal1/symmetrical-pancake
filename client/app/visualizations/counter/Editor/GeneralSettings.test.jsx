import { after } from "lodash";
import React from "react";
import enzyme from "enzyme";

import getOptions from "../getOptions";
import GeneralSettings from "./GeneralSettings";

function findByTestID(wrapper, testId) {
  return wrapper.find(`[data-test="${testId}"]`);
}

function mount(options, done) {
  const data = {
    columns: [
      { name: "a", type: "number" },
      { name: "b", type: "number" },
    ],
    rows: [{ a: 123, b: 987 }],
  };

  options = getOptions(options, data);
  return enzyme.mount(
    <GeneralSettings
      visualizationName="Test"
      data={data}
      options={options}
      onOptionsChange={changedOptions => {
        expect(changedOptions).toMatchSnapshot();
        done();
      }}
    />
  );
}

describe("Visualizations -> Counter -> Editor -> General Settings", () => {
  test("Changes Counter Label", done => {
    const el = mount({}, done);

    findByTestID(el, "Counter.CounterLabel")
      .last()
      .find("input")
      .simulate("change", { target: { value: "Custom Counter Label" } });
  });

  test("Changes Number Format", done => {
    // we will perform 3 actions, so call `done` after all of them completed
    const el = mount({}, after(3, done));

    findByTestID(el, "Counter.NumberFormat")
      .last()
      .find("input")
      .simulate("change", { target: { value: "0,0.0000" } });

    findByTestID(el, "Counter.DecimalCharacter")
      .last()
      .find("input")
      .simulate("change", { target: { value: "-" } });

    findByTestID(el, "Counter.ThousandsSeparator")
      .last()
      .find("input")
      .simulate("change", { target: { value: "/" } });
  });
});
