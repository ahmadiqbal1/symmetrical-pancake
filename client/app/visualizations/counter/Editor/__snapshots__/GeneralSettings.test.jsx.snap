// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualizations -> Counter -> Editor -> General Settings Changes Counter Label 1`] = `
Object {
  "counterLabel": "Custom Counter Label",
}
`;

exports[`Visualizations -> Counter -> Editor -> General Settings Changes Number Format 1`] = `
Object {
  "numberFormat": "0,0.0000",
}
`;

exports[`Visualizations -> Counter -> Editor -> General Settings Changes Number Format 2`] = `
Object {
  "stringDecChar": "-",
}
`;

exports[`Visualizations -> Counter -> Editor -> General Settings Changes Number Format 3`] = `
Object {
  "stringThouSep": "/",
}
`;
